src/ensemble_scheduler/ensemble_scheduler.cc

```diff
@@ -1033,6 +1040,57 @@ EnsembleContext::ReshapeTensorDims(
   return res;
 }
 
+// Caching function
+void
+EnsembleContext::CacheEnsembleTopLevelRequest(
+    std::unique_ptr<InferenceResponse>& response)
+{
+  const std::string key = request_tracker_->Request()->CacheKey();
+  const bool is_key_set = request_tracker_->Request()->CacheKeyIsSet();
+
+#ifdef TRITON_ENABLE_STATS
+  info_->ensemble_end_ns = CaptureTimeNs();
```

Comment:
Are we double counting things here if the "ensemble_end" time is captured after lookup has already finished, and then summing them together? Can you walk me through an example of the time spent on each stage, and confirm if the numbers make sense together?

Reply:
1. Enqueue function calls Cache LookUp function. Time spent in lookup is captured there. If cache hit, only lookup time is stored and request is released.
2. In case of a cache hit, ensemble times are calculated just until cache insertion.
Cache insertion time is calculated separately.
I moved 
#ifdef TRITON_ENABLE_STATS
  info_->ensemble_start_ns = CaptureTimeNs();
  to line 1423 to execute just after cache lookup did not find the request.